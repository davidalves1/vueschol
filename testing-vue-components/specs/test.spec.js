import { mount, shallowMount } from '@vue/test-utils'
import TestComponent from '@/test.vue'
import List from '@/list.vue'

describe('TestComponent', () => {
  it('should console test component', () => {
    console.log(TestComponent)
  })

  it('should mount component corretly', () => {
    const wrapper = mount(TestComponent, {
      propsData: {
        value: 'David Alves'
      }
    })

    // é possível remover o .html() do wrapper no expect
    // pq o snapshotSerializers está configurado no jest.config.js
    expect(wrapper).toMatchSnapshot()
  })
})

describe('ListComponent', () => {
  it('should show list component corretly', () => {
    const wrapper = mount(List)
    const movies = wrapper.vm.marvelMovies
    wrapper.setData({ marvelMovies: [...movies, 'EndGame'] })

    expect(wrapper).toMatchSnapshot()
  })
})
