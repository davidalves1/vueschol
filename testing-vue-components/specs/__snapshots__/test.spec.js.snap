// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent should show list component corretly 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
  </ul>
</div>
`;

exports[`TestComponent should mount component corretly 1`] = `
<div>
  <h1>Hello David Alves</h1>
</div>
`;
